packages/xen-api/src/index.js

```diff
@@ -468,12 +468,11 @@ export class Xapi extends EventEmitter {
       body,
       headers,
       rejectUnauthorized: !this._allowUnauthorized,
-
       // this is an inactivity timeout (unclear in Node doc)
       timeout: this._httpInactivityTimeout,
-
       // Support XS <= 6.5 with Node => 12
       minVersion: 'TLSv1',
+      agent: this.httpAgent,
```

Comment:
I think that since the agent is bound here, it is not needed anymore on line 489 (but github don't let me make a code change suggestion on a line not touched by the PR )

Reply:
Well, that's weird because without binding agent in line 475, it is not using the agent on line 489.

After some testing, the same issue occurs when applying rolling backup, it fails with connection timeout.

Reply:
Hi @fbeauchamp the commit on line 489 was made by your user. After analyzing the code, the agent attribute is supposed to be consumed correctly without my PR. Did I missed something?

Reply:
Sorry to be so verbose. I think I found the problem: there's a ternary and maybe when creating cloudconfig, it is using the "else" which does not use httpAgent bound in line 489.
To be sure, I'll remove the duplicated assignment and configure it on line 475 ok?